experimental/devices/hx711/hx711.go

```diff
@@ -0,0 +1,180 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package hx711 implements an interface to the HX711 analog to digital converter.
+//
+// Datasheet: https://www.mouser.com/ds/2/813/hx711_english-1022875.pdf
+package hx711
+
+import (
+	"errors"
+	"time"
+
+	"periph.io/x/periph/conn/gpio"
+)
+
+var (
+	// TimeoutError is returned from Read and ReadAveraged when the ADC took too
+	// long to indicate data was available.
+	TimeoutError = errors.New("timed out waiting for HX711 to become ready")
+)
+
+// InputMode controls the voltage gain and the channel multiplexer on the HX711.
+// Channel A can be used with a gain of 128 or 64, and Channel B can be used
+// with a gain of 32.
+//
+// Changing the InputMode on an HX711 will only take effect *after* the next
+// read.
+type InputMode int
+
+const (
+	CHANNEL_A_GAIN_128 InputMode = 1
+	CHANNEL_A_GAIN_64  InputMode = 3
+	CHANNEL_B_GAIN_32  InputMode = 2
+
+	dataBits         = 24
+	readPollInterval = 50 * time.Millisecond
+)
+
+type HX711 struct {
+	InputMode InputMode
+
+	clk     gpio.PinOut
+	data    gpio.PinIn
+	useEdge bool
+	done    chan struct{}
+}
+
+// New creates a new HX711 device.
+func New(clk gpio.PinOut, data gpio.PinIn) (*HX711, error) {
+	// Try enabling edge detection on the data pin.
+	var useEdge bool
+	if err := data.In(gpio.PullDown, gpio.FallingEdge); err != nil {
+		if err := data.In(gpio.PullDown, gpio.NoEdge); err != nil {
```

Comment:
I'm in the process of adding gpioutil.PollEdge() for software based polling for edge. It's currently at an incorrect location, the end location would be conn/gpio/gpioutil.
https://periph.io/x/periph/experimental/devices/gpioutil#PollEdge

Can you use it and confirm that the code works properly? I didn't promote to stable yet because I need folks to test it and find bugs. :) It's actually more tricky than it looks like; https://github.com/google/periph/blob/master/experimental/devices/gpioutil/polledge.go

Reply:
Cool!  I'll give this a try tonight!

Reply:
Done!  It seems to work!
There's a race condition though.  I want to wait for the pin to be low, regardless of the current level.  But if I do:

    if pin.Read() != gpio.Low {
      pin.WaitForEdge(timeout)
    }

and it goes low between my call to pin.Read() and WaitForEdge's curr := p.PinIO.Read(), then it'll miss it and wait for the timeout.
I think I really wait a WaitForLevel()!